ui/src/test/java/edu/wpi/grip/ui/codegeneration/tools/PythonPipelineInterfacer.java

```diff
@@ -72,19 +75,30 @@ public PythonPipelineInterfacer(String className) {
 
   @Override
   public void setNumSource(int num, Number value) {
-    str.append("source = ").append(value.doubleValue()).append(newLine).append("pipe.set_source")
-        .append(num).append("(source)").append(newLine);
+    sourceNames.add("source" + num);
+    str.append("source").append(num).append(" = ").append(value.doubleValue())
+        .append(newLine);
   }
 
   @Override
   public void setMatSource(int num, File img) {
-    str.append("source = cv2.imread(").append(pyPath(img.getAbsolutePath())).append(')')
-        .append(newLine).append("pipe.set_source").append(num).append("(source)").append(newLine);
+    sourceNames.add("source" + num);
+    str.append("source").append(num)
+        .append(" = cv2.imread(").append(pyPath(img.getAbsolutePath())).append(')')
+        .append(newLine);
   }
 
   @Override
   public void process() {
-    str.append("pipe.process()").append(newLine);
+    str.append("pipe.process(");
+    for (int i = 0; i < sourceNames.size(); i++) {
+      str.append(sourceNames.get(i));
+      if (i < sourceNames.size() - 1) {
+        str.append(", ");
+      }
+    }
```

Comment:
Apache has you covered on this: `StringUtils.join()`
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#join-java.util.Iterator-java.lang.String-